### PR TITLE
Enable scheduler and higher starting LR to avoid plateau

### PR DESCRIPTION
**Pull request by michaelmckinsey1**
Use `ExponentialLR` scheduler to start at a higher LR to avoid getting stuck at `val_dice_score=0.5`. Decreases number of epochs required on the order of 2x. `main` branch currently uses a default config with a constant LR.

- [x] starting_learning_rate, with gamma and min_learning_rate added
- [x] Exclude background class 0 from dice calculation, so `val_dice_score` is not boosted by background class.

main vs PR#53:

1node:
- scale6 - 329 vs 106 epochs (3.1x)
- scale7 - 343 epochs vs 208 epochs (1.6x)
- scale8 - 406 epochs vs 296 epochs (1.4x)

2nodes:
- scale6 -  616 vs 240 epochs (2.6x)

### CODE REVIEW

File: ScaFFold/utils/evaluate.py [hunk 1]
```diff
@@ -118,11 +123,11 @@ def evaluate(
             dice_score_probs = compute_sharded_dice(
                 mask_pred_probs, mask_true_onehot, spatial_mesh
             )
-            dice_loss_curr = 1.0 - dice_score_probs.mean()
```
Comment (michaelmckinsey1) [inline ScaFFold/utils/evaluate.py:121]: was background + foreground -> now foreground

File: ScaFFold/utils/config_utils.py [hunk 2]
```diff
@@ -61,7 +61,18 @@ def __init__(self, config_dict):
...
+        self.scale_reference = config_dict["scale_reference"]
+        self.scale_reference_starting_learning_rate = config_dict[
+            "scale_reference_starting_learning_rate"
+        ]
+        self.scale_learning_rate_factor = config_dict["scale_learning_rate_factor"]
+        self.starting_learning_rate = (
+            self.scale_reference_starting_learning_rate
+            * self.scale_learning_rate_factor
+            ** (self.problem_scale - self.scale_reference)
+        )
```
Comment (michaelmckinsey1) [inline ScaFFold/utils/config_utils.py:70]: Noticed instability at higher problem scales that is fixed by lowering the LR

Comment (michaelmckinsey1) [reply ScaFFold/utils/config_utils.py:70]: Noticed this is correlated with turning on AMP

Comment (michaelmckinsey1) [reply ScaFFold/utils/config_utils.py:70]: no AMP
image
vs AMP
image

Comment (michaelmckinsey1) [reply ScaFFold/utils/config_utils.py:70]: probably should just omit this logic and avoid using AMP

**Comment by michaelmckinsey1** [comment]
Superseded  by #59 and #61
